transports/jms/src/main/java/org/mule/transport/jms/JmsConnector.java

```diff
@@ -1466,4 +1472,14 @@ protected Object getOperationResourceFactory()
     {
         return getConnection();
     }
+
+    /**
+     * Obtain the shared timer instance used by this JMS Connector for performing timeout of async responses.
+     *
+     * @return timer
+     */
+    public Timer getResponseTimeoutTimer()
```

Comment:
Can you hide the timer form the caller and provide a method to do the scheduling?